src/platform_impl/linux/x11/util/modifiers.rs

```diff
@@ -34,8 +34,7 @@ pub struct ModifierKeymap {
 
 #[derive(Clone, Debug, Default)]
 pub struct ModifierKeyState {
-    // Contains currently pressed modifier keys and their corresponding modifiers
-    keys: HashMap<ffi::KeyCode, Modifier>,
```

Comment:
And this is removing duplicate state. Looking at the history, originally there was only `keys`, then `state` was added. I think having only state is simpler.